force-app/main/default/classes/CandidateGradeController.cls

```diff
@@ -3,8 +3,12 @@ public with sharing class CandidateGradeController {
     }
     
     @AuraEnabled
-    public static CandidateGradeInfo getCandidateGrade(Id contactId) {
-        Contact currentContact = [SELECT Name, Salary__c, NextAvailableDate__c FROM Contact where Id = :contactId];
+    public static CandidateGradeInfo getCandidateGrade(String contactId) { 
+        
+        // Build the query
+        String selectQueryString = 'SELECT Name, Salary__c, NextAvailableDate__c FROM Contact where Id = \'' + contactId + '\'';
+        Contact currentContact = Database.query(selectQueryString);
```

Comment:
This statement may be vulnerable to SOQL/SOSL injection. Please sanitize any unsafe expressions using `String.escapeSingleQuotes`.